File: src/main/java/pl/bartlomiejstepien/armaserverwebgui/domain/purge/DataPurgeService.java
package pl.bartlomiejstepien.armaserverwebgui.domain.purge;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import pl.bartlomiejstepien.armaserverwebgui.domain.server.mission.MissionService;
import pl.bartlomiejstepien.armaserverwebgui.domain.server.mod.ModPresetService;
import pl.bartlomiejstepien.armaserverwebgui.domain.server.mod.ModService;

@Slf4j
@Service
@RequiredArgsConstructor
public class DataPurgeService
{
    private final ModService modService;
    private final ModPresetService modPresetService;
    private final MissionService missionService;

    public void purgeMissions(boolean deleteFiles)
    {
        log.info("Deleting all missions. Include files: {}", deleteFiles);
        this.missionService.deleteAllMissions(deleteFiles);
    }

    public void purgeMods(boolean deleteFiles)
    {
        log.info("Deleting all mods. Include files: {}", deleteFiles);
        this.modService.deleteAllMods(deleteFiles);
    }

    public void purgeModPresets()
    {
        log.info("Deleting all mod presets.");
        this.modPresetService.deleteAllPresets();
    }
}

File: src/web-ui/src/app/service/data-purge.service.ts
import { inject, Service } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { API_BASE_URL } from "@environments/environment";
import { Observable } from "rxjs";

@Service()
export class DataPurgeService {
  private readonly httpClient = inject(HttpClient);

  public purgeModPresets(): Observable<void> {
    return this.httpClient.delete<void>(`${API_BASE_URL}/settings/purge/mod-presets`);
  }

  public purgeMissions(deleteFiles: boolean): Observable<void> {
    return this.httpClient.delete<void>(`${API_BASE_URL}/settings/purge/missions`, {
      params: {
        "delete-files": deleteFiles
      }
    });
  }

  public purgeMods(deleteFiles: boolean): Observable<void> {
    return this.httpClient.delete<void>(`${API_BASE_URL}/settings/purge/mods`, {
      params: {
        "delete-files": deleteFiles
      }
    });
  }
}
